tree/ntuple/v7/src/RNTupleProcessor.cxx

```diff
@@ -231,26 +242,33 @@ ROOT::Experimental::NTupleSize_t ROOT::Experimental::RNTupleChainProcessor::Conn
    return fPageSource->GetNEntries();
 }
 
-ROOT::Experimental::NTupleSize_t ROOT::Experimental::RNTupleChainProcessor::Advance()
+ROOT::Experimental::NTupleSize_t ROOT::Experimental::RNTupleChainProcessor::LoadEntry(NTupleSize_t entryNumber)
 {
-   if (fLocalEntryNumber == kInvalidNTupleIndex)
-      return kInvalidNTupleIndex;
+   NTupleSize_t localEntryNumber = entryNumber;
+   size_t currentNTuple = fCurrentNTupleNumber;
 
-   if (fLocalEntryNumber >= fPageSource->GetNEntries()) {
-      do {
-         if (++fCurrentNTupleNumber >= fNTuples.size()) {
-            return kInvalidNTupleIndex;
-         }
-         // Skip over empty ntuples we might encounter.
-      } while (ConnectNTuple(fNTuples.at(fCurrentNTupleNumber)) == 0);
+   while (localEntryNumber >= fInnerNEntries.at(currentNTuple)) {
+      localEntryNumber -= fInnerNEntries.at(currentNTuple);
```

Comment:
I'm not a big fan of all these repeated `.at`s: I would argue it's better to check the condition once at the start of the function (throwing a helpful error message if the number is out of bounds) and then use the regular `[]` access throughout the rest of the function.